src/Docker/DockerIntegrationBuilder.php

```diff
@@ -0,0 +1,213 @@
+    {
+        $phpVersion = $repository->get(self::PHP_VERSION);
+        $dbVersion = $repository->get(self::DB_VERSION);
+        $nginxVersion = $repository->get(self::NGINX_VERSION);
+
+        $services = [
+            'version' => '2',
+            'services' => [
+                'fpm' => $this->serviceFactory->create(
+                    ServiceFactory::SERVICE_FPM,
+                    $phpVersion,
+                    [
+                        'ports' => [9000],
+                        'links' => [
+                            'db',
+                        ],
+                        'volumes' => [
+                            $this->getMagentoVolume(false)
+                        ],
+                        'volumes_from' => [
+                            'appdata',
+                        ],
+                        'env_file' => [
+                            './docker/global.env',
+                            './docker/composer.env',
+                        ],
+                    ]
+                ),
+                'db' => $this->serviceFactory->create(
+                    ServiceFactory::SERVICE_DB,
+                    $dbVersion,
+                    [
+                        'ports' => [3306],
+                        'volumes' => [
+                            '/var/lib/mysql',
+                        ],
+                        'environment' => [
+                            'MYSQL_ROOT_PASSWORD=magento2',
+                            'MYSQL_DATABASE=magento2',
+                            'MYSQL_USER=magento2',
+                            'MYSQL_PASSWORD=magento2',
+                        ],
+                    ]
+                ),
+                'web' => $this->serviceFactory->create(
+                    ServiceFactory::SERVICE_NGINX,
+                    $nginxVersion,
+                    [
+                        'ports' => [
+                            '8030:80',
+                        ],
+                        'links' => [
+                            'fpm',
+                            'db',
+                        ],
+                        'volumes' => [
+                            $this->getMagentoVolume(false)
+                        ],
+                        'volumes_from' => [
+                            'appdata',
+                        ],
+                        'env_file' => [
+                            './docker/global.env',
+                            './docker/composer.env',
+                        ],
+                    ]
+                ),
+            ],
+            'volumes' => [
+                'magento' => []
+            ]
+        ];
+
+        $services['services']['build'] = $this->getCliService(
+            $phpVersion,
+            false,
+            ['db'],
+            'build.magento2.docker'
+        );
+        $services['services']['deploy'] = $this->getCliService(
+            $phpVersion,
+            false,
```

Comment:
why is it false?

Reply:
Still under development, Docker engine failing for some reason

Reply:
@shiftedreality Maybe you need to create a task to remember to change this argument.